litgpt/tokenizer.py

```diff
@@ -96,18 +96,23 @@ def encode(
         elif self.backend == "sentencepiece":
             tokens = self.processor.encode(string)
         else:
-            raise RuntimeError
+            raise RuntimeError(f"`{self.backend}` is not supported.")
+        if tokens is None:
+            raise ValueError("`self.processor` returned tokens of None value.")
+
         if bos or (bos is None and self.use_bos):
             bos_id = self.bos_id
```

Comment:
Is this `bos_id = self.bos_id` assignment necessary here since we don't use it below anyway.

Reply:
Good question.
Well, in theory, it should improve the performance, since we don't need to retrieve this attribute every single time.
And I guess that was the idea when Carlos wrote the code, but yeah, I think we can drop it :)

Reply:
I think it's not a big deal regarding performance. But we can keep it, but then maybe using it consistently. Right now it's just a bit weird.